src/validation.h

```diff
@@ -194,12 +194,22 @@ void UnlinkPrunedFiles(const std::set<int>& setFilesToPrune);
 /** Prune block files up to a given height */
 void PruneBlockFilesManual(int nManualPruneHeight);
 
-/** (try to) add transaction to memory pool
- * plTxnReplaced will be appended to with all transactions replaced from mempool
- * @param[out] fee_out optional argument to return tx fee to the caller **/
-bool AcceptToMemoryPool(CTxMemPool& pool, TxValidationState &state, const CTransactionRef &tx,
-                        std::list<CTransactionRef>* plTxnReplaced,
-                        bool bypass_limits, bool test_accept=false, CAmount* fee_out=nullptr) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
+/** Per-transaction result from trying to accept a transaction to the memory pool. */
+struct MempoolAcceptResult {
+    MempoolAcceptResult(const CTransactionRef& ptx) : txid(ptx->GetHash()) {}
+
+    uint256 txid;
```

Comment:
in commit c97fae1011c4bd376898af73576756f163eeaf58:

This member seems currently unused? Also, what is the point of copying the (already cached) txid once more? If you need a reference to the tx, maybe store the `CTransactionRef`, but that seems redundant, because the caller is already aware of the tx.

Reply:
Right, it's not used in https://github.com/bitcoin/bitcoin/commit/c97fae1011c4bd376898af73576756f163eeaf58. I ended up needing it for multi-accept on the failure case to indicate which tx failed. It might fit better in https://github.com/bitcoin/bitcoin/pull/20833/commits/2f36f0743158817635a39f7efb60541fe0c4b31d

Reply:
Moved it to the relevant commit. Putting a `CTransactionRef` in the `MempoolAcceptResult` when returning failure so that we know which tx failed; we need it to get the txid and wtxid in `testmempoolaccept`.